src/parse.rs

```diff
@@ -534,7 +544,9 @@ where
         }
     }
 
-    fn try_consume_bracket_class_atom(&mut self) -> Result<Option<ClassAtom>, Error> {
+    fn try_consume_bracket_class_atom(
+        &mut self,
+    ) -> Result<Option<(ClassAtom, Option<ClassAtom>)>, Error> {
```

Comment:
This is a complicated return type. Here's two alternatives for simplifying it:

1. When encountering "\c" in a bracket, instead of returning both, unread the "c" and only return the backslash. The next iteration through the loop will then pick up the "c".
2. Instead return a `Result<Vec<ClassAtom>, Error>`

I think the first will end up quite clean assuming it works.

Reply:
That sounds good, I will try to change it to the first if it works.

Reply:
Works well. I implemented it in #88.